### DISCUSSION TITLE
[Feature Request] Janitorial Science Device: Mess Scanner

### DISCUSSION THREAD

**Discussion by eurtod** [unitystation/unitystation#9573] [Ideas]
### There is no other issue with the same feature request.

- [X] I have searched the existing issues

### Why is this feature needed/good for the game?

Miners have a piece of technology that allows them to scan for ores, when it is activated it highlights every ore nearby through walls and darkness. Something similar would enable janitors to have an easier time finding messes, knowing what departments to ask for access to clean, and other benefits. It would also add more things to Janitorial science which I think more technology for science to research and provide to departments is always a plus.

### Describe your feature in detail.

After researching and creating the Mess Scanner in a prolathe or autolathe (whichever is the right one), the scanner can be clicked and would highlight all spills, footprints, items considered trash, etc. It would highlight everything that is in the FOV no matter if it's behind walls or obscured by darkness similar to the mining scanner, then after a few seconds everything would stop being highlighted until the scanner is activated again.

### User stories

As a janitor, I want to be able to know where messes are without having to ask to enter a department or room first.
As a janitor, I want to be able to know if I missed any spots when cleaning an area.
As a scientist, I want to be able to provide more useful tools to the janitor so they can keep the station cleaner.,
As a scientist, I want to be able to unlock a Mess Scanner under Janitorial science on the tech web.
As a player, I want to be able to create a Mess scanner from the prolathe or autolathe after it is unlocked from the tech web.
As a player, I want to be able to click on the Mess Scanner and to have all messes that I can and can't see through walls and such be highlighted for a few seconds and then to disappear.
As a janitor, I want the Mess Scanner to highlight spills, puddles, powders, footprints, trash like broken glass and empty boxes, etc. like a mining scanner when I click on it.

### Describe alternatives you've considered.

I considered having this to be an innate ability for Janitors to have and would work like Mime spells but decided it wouldn't make much sense.

**Comment by eurtod** [comment]
Why was this moved to a discussion? It's a feature request?

**Reply by eurtod** [reply]
@corp-0 is this right place for this? in discussions??

**Reply by corp-0** [reply]
yeah, no. @PerfectTangent in this case, since it has clear requirements, it should be an issue. The discussion is being held on discord.

**Reply by corp-0** [reply]
the point of discussions, wherever they originated, are to end with an issue with clear requirements.

**Reply by PerfectTangent** [reply]
> yeah, no. @PerfectTangent in this case, since it has clear requirements, it should be an issue. The discussion is being held on discord.

that is a departure from how we have handled _novel_ ideas and contradicts the last 2 years of both you and me moving all suggestions out of issues and into discussions. We don't want people to submit ideas as issues until we've determined that they are ideas worth including in the game, then they get upgraded to be an issue.
